synapse/tools/superhash.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+synapse - superhash.py
+Created on 4/26/17.
+
+Generate the hashsets and guid (superhash) for a file.
+"""
+# Stdlib
+import argparse
+import json
+import logging
+import os
+import sys
+# Custom Code
+import synapse.axon as s_axon
+import synapse.lib.output as s_output
+
+log = logging.getLogger(__name__)
+
+
+def compute_hashes(fp):
+    """
+    Compute the superhash information for a file path
+    :param fp: Path to compute
+    :return: Tuple containing the superhash (guid) and other hashes for the file.
+    """
+    hs = s_axon.HashSet()
+    with open(fp, 'rb') as f:
+        guid, hashd = hs.eatfd(fd=f)
+    return guid, hashd
+
+
+def main(argv, outp=None):
+
+    if outp == None:  # pragma: no cover
+        outp = s_output.OutPut()
+
+    p = makeargpaser()
+    opts = p.parse_args(args=argv)
+
+    results = []
+    for fp in opts.input:
+        try:
+            guid, hashd = compute_hashes(fp=fp)
+        except:
+            outp.printf('Failed to compute superhash for {}'.format(fp))
+        else:
+            results.append((fp, guid, hashd))
+
+    if opts.ingest:
+        l = []
+        for fp, guid, hashd in results:
+            hashd['name'] = os.path.basename(fp)
+            d = {"props": hashd}
+            hl = [guid, d]
+            l.append(hl)
+        if len(l) == 1:
+            l = l[0]
+        outp.printf(json.dumps(l, sort_keys=True, indent=2))
+    else:
+        for fp, guid, hashd in results:
+            outp.printf('Superhash for: {}'.format(fp))
+            hashd['guid'] = guid
+            keys = list(hashd.keys())
+            keys.sort()
+            for key in keys:
+                value = hashd.get(key)
+                outp.printf('{}\t{}'.format(key, value))
+
+    return 0
+
+def makeargpaser():
+    parser = argparse.ArgumentParser(description="Compute the guid and hashes for a file.")
```

Comment:
direct use of argparse will prevent this from being used in code that doesn't want to exit the process on syntax error....  do we ever want to use this command or main() function as anything other than a stand alone cli?  ( also, makes implementing tests which fail on purpose difficult since argparse raises SystemExit )